__tests__/app.test.js

```diff
@@ -0,0 +1,58 @@
+const request = require("supertest");
+const app = require("../app");
+const db = require("../db/connection");
+const data = require("../db/data/test-data/index");
+const seed = require("../db/seeds/seed");
+
+beforeEach(() => {
+  return seed(data);
+});
+afterAll(() => {
+  return db.end();
+});
+
+describe("GET /not-a-path", () => {
+  test("404: responds with a 404 if path not found with a user message to confirm reason for error", () => {
+    return request(app)
+      .get("/not-a-path")
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("path not found");
+      });
+  });
+});
+
+describe("GET /api/topics", () => {
+  describe("Functionality", () => {
+    test("200: should return an array of objects with keys of slug and description", () => {
+      return request(app)
+        .get("/api/topics")
+        .expect(200)
+        .then(({ body }) => {
+          expect(Array.isArray(body.topics)).toBe(true);
+          body.topics.forEach((topic) => {
```

Comment:
Nice job, but if you're going to forEach through an array you should check the length of the array first as if an empty array came back it would cause a false pass for this test.